includes/shell/utility/split.hpp

```diff
@@ -0,0 +1,68 @@
+
+#pragma once
+
+#include "types.hpp"
+
+#include <string>
+#include <string_view>
+
+namespace shell {
+
+class ForwardLazySpliterator; // forward decleration
+
+/*!
+ * @brief Lazily split strings.
+ */
+class LazySplit {
+	const string source;    /*!< -.*/
```

Comment:
What does the `-.` mean in the comments?

Reply:
Hahaha yeah it is what we use at work for saying the variable or function name is self explanatory so no comment is needed but our automatic documentation generator does require everything to be documented so we use '-.'. Anyhow it is just a habit, ill remove them if its confusing😁

Reply:
Requiring everything to be documented and then having this sounds like a nightmare...

Reply:
You are not wrong, i got a lot of shit on prs about not doing it😂